### PR TITLE
WIP: Add --display to run|install

### PR DESCRIPTION
**Pull request by sallyom**
Signed-off-by: Sally O'Malley <somalley@redhat.com>

### CODE REVIEW

**Comment by sallyom** [comment]
list of test cmds i've run with output: http://fpaste.org/246219/42462114/

**Comment by rhatdan** [comment]
Why do you need to check self.args.display all over the place?

**Comment by rhatdan** [comment]
This should only be for run and install.  Remove all other checks.

**Comment by sallyom** [comment]
there were a few extra checks in there :)

**Comment by sallyom** [comment]
note WIP, I have the integration test running i'm still working on it, for example, when I change the LABEL RUN in the Dockerfile and leave the expected outcome the same in test_display.sh I expect the test to fail but it passes. @willmtemple please review if you have time, thanks :)

**Comment by rhatdan** [comment]
Need updated man pages.

**Comment by sallyom** [comment]
@willmtemple , I implemented your fb with the tests, thanks again.

**Comment by sallyom** [comment]
added def display(self, cmd)

**Comment by sallyom** [comment]
implemented fb, updated man pages

**Comment by rhatdan** [comment]
LGTM.

@willmtemple @cgwalters 
Could you guys take a look.

**Comment by willmtemple** [comment]
@rhatdan @sallyom I'm not sure we should include an image that might actually manipulate the host as a test. This sounds like in order to test `atomic run` on this image, I would have to actually start an httpd on my system. Could we construct a toy image instead that would test the core functionality in the same way?

**Comment by rhatdan** [comment]
@willmtemple Good catch.  

Lets change the Dockerfile to only have the labels, to not do anything other then echo something.

LABEL INSTALL "docker run ...\${image} echo I am the install label"
LABEL RUN "docker run ...\${image} echo I am the run label"
LABEL UNINSTALL "docker run ...\${image} echo I am the uninstall label"

**Comment by willmtemple** [comment]
@rhatdan @sallyom That sounds good. We should try to work the --opt<n> arguments, and any settable environment variables into those labels.

**Comment by sallyom** [comment]
ok, will do

**Comment by sallyom** [comment]
Done, let me know if i covered everything with the LABEL RUN, LABEL INSTALL

**Comment by rhatdan** [comment]
I think you should put legitimate commands into your Dockerfile, these commands should cause the tests to blow up.  You should also check with out the --display to make sure --display did not break anything.

File: tests/test-images/Dockerfile.1 [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
+FROM centos
+MAINTAINER "Sally O'Malley <somalley at redhat dot com>
+ENV container docker
 
 LABEL "Name"="atomic-test-1"
+
+LABEL INSTALL "docker run --rm --spc \$SUDO_UID:\$SUDO_GID --privileged \$OPT1 FOO1 \$OPT2 FOO2 -v /:/host -e HOST=/host -e LOGDIR=${LOGDIR} -e CONFDIR=${CONFDIR} -e \${DATADIR} --name NAME IMAGE  echo I am the install label /bin/install.sh"
+LABEL RUN "docker run -t -i --rm --spc \$SUDO_UID:\$SUDO_GID \$OPT1 FOOBAR1 \$OPT2 FOOBAR2 \$OPT3 FOOBAR3 -v LOGDIR:/var/log -v DATADIR:/var/lib --name NAME IMAGE echo I am the run label run.sh"
+
+EXPOSE 80
```
Comment (willmtemple) [inline tests/test-images/Dockerfile.1]: I would drop this line, as well as the --spc and --privileged options in the run and install labels.

**Comment by sallyom** [comment]
@willmtemple carriage returns (^M) were messing up my test output, finally got past it.
@rhatdan I added the run,install tests and checked against output of 'docker logs container'

**Comment by willmtemple** [comment]
@sallyom @rhatdan If we get rid of privileged options (`--cap-add=SYS_ADMIN`, `--net=host`, `--ipc=host`) in the dockerfile, then it LGTM. All tests are passing on my end.

**Comment by rhatdan** [comment]
Yes remove them, and then we can merge.

**Comment by sallyom** [comment]
OK, done

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**